articles/150_roslaunch.md

```diff
@@ -0,0 +1,821 @@
+- use groups to apply settings (e.g. remappings) to collections of nodes/processes/included launch files
+  - also use groups with namespaces to form hierarchies
+- portability through abstraction of operating system concepts, e.g. environment variables
+- utilities to locate files on the filesystem in a relocatable and portable way, e.g. `$(find <package_name>)`
+
+That covers most of the features and design goals of `roslaunch` from ROS 1, but in the next subsection we'll discuss what is different for the launch system in ROS 2 due to changes in ROS 2 and how it might improve on the launch system from ROS 1.
+
+### Differences in ROS 2
+
+One of the objectives of the launch system in ROS 2 is to emulate the features of the launch system in ROS 1, but due to architectural changes in ROS 2, some of the features, goals, and terminology need to change.
+
+#### Relationship Between Nodes and Processes
+
+In ROS 1, there could only ever be one node per process and so the goals of `roslaunch` from ROS 1 reflect that by using "ROS nodes" and "processes" almost interchangeably.
+
+Even for the ROS 1 feature called 'nodelet' (where you could emulate having more than one node per process), the conceptual mapping from node or nodelet to process was preserved by proxy processes.
+For example, you would run a "NodeletManager" and then run a process for each nodelet you wanted to run in that manager.
+This allowed nodelet's which exited to be detected by `roslaunch` from ROS 1, as well as allowing them to respond to signals that it sent to the proxy process.
+
+Since you can have many nodes per process in ROS 2, it is no longer necessary to conflate nodes and processes.
+Due to this, the design and documentation for the launch system in ROS 2 will need to be clearer when talking about processes and nodes.
+Additionally, the way that configuration (e.g. parameters and remappings) get passed to nodes by the launch system needs to be adapted, though this part overlaps with the design documents for static remapping[^static_remapping] and for parameters[^parameters].
+
+Also, since there can be multiple nodes per process, shutting down a node no longer always means sending a unix signal to a single process.
+Other mechanisms might need to be used to have more granular shutdown control in multi-node processes.
+
+#### Launching Nodes (Processes) Remotely and Portability
+
+The launch system in ROS 1 only really ever was supported on Linux and other Unix-like operating systems like BSD and macOS.
+These machines all have SSH, which is the mechanism which is specifically called out to be used when launching processes on remote machines.
+It also played a role in defining what you specified and how when configuring `roslaunch` from ROS 1 to be able to launch processes on remote machines.
+
+In ROS 2, Windows has been added to the list of targeted platforms, and as of the writing of this document it does not support SSH natively.
```

Comment:
Would requiring the execution of a launcher "worker" process on each additional available machine be a good solution for this? The worker could register its hostname, ip etc. (via a normal ros topic maybe?).

Would people be OK with each additional computer requiring some sort of ros2 launch client running? There could be some system that allows waiting for worker clients, handling events on an expected 2nd(+) computer failing to be present (e.g. timeout, connection loss).